etcdserver/api/membership/store.go

```diff
@@ -72,8 +72,20 @@ func mustSaveClusterVersionToBackend(be backend.Backend, ver *semver.Version) {
 
 	tx := be.BatchTx()
 	tx.Lock()
-	defer tx.Unlock()
 	tx.UnsafePut(clusterBucketName, ckey, []byte(ver.String()))
+	tx.Unlock()
```

Comment:
Is there a reason we changed line 75? In case UnsafePut fails, the deferred statement seems better.

Reply:
I think there is only one line between `Lock()` and `Unlock()` so that it is not necessary to use `defer()`? If UnsafePut fails, why using defer is better?

Reply:
I meant if UnsafePut panics, defer ensures that the tx is unlocked.

Reply:
I don't think `defer` can work well if something panic happens. My understanding is that `defer()` works after the surrounding function returns. If the underlying function panics, then `defer` statement will never be called.  If we want to ensure handling unexpected panic before `unlock()`, I think we should use `recover()` like this.
```
defer func() {
        if err := recover(); err != nil {
            log.Println("panic occurred:", err)
        }
		tx.Unlock()
    }()
```